npm/react/cypress/component/advanced/renderless/mouse-movement.jsx

```diff
@@ -1,52 +1,44 @@
 // https://medium.com/@pierrehedkvist/renderless-components-in-react-8d663746314c
-import React from 'react'
+// eslint-disable-next-line no-unused-vars
+import React, { useState, useEffect } from 'react'
 import PropTypes from 'prop-types'
 
-export default class MouseMovement extends React.Component {
-  constructor (props) {
-    console.log('MouseMovement constructor')
-    super(props)
-    this.state = {
-      timer: undefined,
-    }
-
-    this.timeout = this.timeout.bind(this)
-    this.onMouseMove = this.onMouseMove.bind(this)
-  }
+export default function MouseMovement ({ onMoved }) {
+  const [timer, setTimer] = useState(undefined)
 
-  componentWillMount () {
-    console.log('MouseMovement componentWillMount')
-    document.addEventListener('mousemove', this.onMouseMove)
-    const timer = setTimeout(this.timeout, 4000)
+  function onMouseMove () {
+    console.log('MouseMovement onMouseMove')
+    clearTimeout(timer)
+    const timerNew = setTimeout(timeout, 4000)
 
-    this.setState({ timer })
+    setTimer(timerNew)
+    onMoved(true)
   }
 
-  componentWillUnmount () {
-    console.log('MouseMovement componentWillUnmount')
-    document.removeEventListener('mousemove', this.onMouseMove)
-    clearTimeout(this.state.timer)
-    this.setState({ timer: undefined })
+  function timeout () {
+    console.log('timeout')
+    clearTimeout(timer)
+    onMoved(false)
   }
 
-  onMouseMove () {
-    console.log('MouseMovement onMouseMove')
-    clearTimeout(this.state.timer)
-    const timer = setTimeout(this.timeout, 4000)
+  useEffect(() => {
+    // Anything in here is fired on component mount.
```

Comment:
There's no dependency array here. Isn't this going to run on every re-render?

Reply:
Saw this in a couple places. Shouldn't the eslint react plugin be catching this?